data-model/src/main/java/io/micronaut/data/model/query/builder/sql/SqlQueryBuilder.java

```diff
@@ -1029,44 +1029,37 @@ protected void appendUpdateSetParameter(StringBuilder queryString, PersistentPro
                         String[] joinColumnNames = resolveJoinTableColumns(associationOwner, associatedEntity, association, identity, associatedEntity.getIdentity(), namingStrategy);
                         String joinTableAlias = joinAliases[i] + joinTableName + "_";
                         String associatedTableName = getTableName(associatedEntity);
-                        target.append(joinTableName)
-                              .append(SPACE)
-                              .append(joinTableAlias)
-                              .append(" ON ")
-                              .append(alias)
-                              .append(DOT)
-                              .append(escape ? quote(getColumnName(associatedId)) : getColumnName(associatedId))
-                              .append('=')
-                              .append(joinTableAlias)
-                              .append(DOT)
-                              .append(joinColumnNames[0])
-                              .append(SPACE)
-                              .append(joinType)
-                              .append(associatedTableName)
-                              .append(SPACE)
-                              .append(joinAliases[i])
-                              .append(" ON ")
-                              .append(joinTableAlias)
-                              .append(DOT)
-                              .append(joinColumnNames[1])
-                              .append('=')
-                              .append(joinAliases[i])
-                              .append(DOT)
-                              .append(escape ? quote(getColumnName(associatedEntity.getIdentity())) : getColumnName(associatedEntity.getIdentity()));
+
+                        StringBuilder join = joinStringBuilder(joinType,
+                                joinTableName,
+                                joinTableAlias,
+                                alias,
+                                escape ? quote(getColumnName(associatedId)) : getColumnName(associatedId),
+                                joinColumnNames[0]);
+                        if (target.indexOf(join.toString()) == -1) {
```

Comment:
since you are converting the builder to a string at this point it makes sense to use the same converted variable to append to the target otherwise `toString()` will be called twice which is not super efficient

Also you can make just use `target.contains(..)` instead

Reply:
`target` is of type `StringBuilder`, it does not have `contains` method.

Reply:
> since you are converting the builder to a string at this point it makes sense to use the same converted variable to append to the target otherwise toString() will be called twice which is not super efficient

I've done the change so that `toString()`  is called only once.